examples/secrets/values-secret.v2.yaml

```diff
@@ -0,0 +1,81 @@
+# NEVER COMMIT THESE VALUES TO GIT (unless your file only uses generated
+# passwords or only points to files)
+
+# Needed to specify the new format (missing version means old version: 1.0 by default)
+version: 2.0
+
+# These are the vault policies to be created in the vault
+# these are used when we let the vault generate the passwords
+# by setting the 'onMissingValue' attribute to 'generate'
+# See https://developer.hashicorp.com/vault/docs/concepts/password-policies
+vaultPolicies:
+  basicPolicy: |
+    length=10
+    rule "charset" { charset = "abcdefghijklmnopqrstuvwxyz" min-chars = 1 }
+    rule "charset" { charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" min-chars = 1 }
+    rule "charset" { charset = "0123456789" min-chars = 1 }
+
+  advancedPolicy: |
+    length=20
+    rule "charset" { charset = "abcdefghijklmnopqrstuvwxyz" min-chars = 1 }
+    rule "charset" { charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" min-chars = 1 }
+    rule "charset" { charset = "0123456789" min-chars = 1 }
+    rule "charset" { charset = "!@#$%^&*" min-chars = 1 }
+
+# This is the mandatory top-level secrets entry
+secrets:
+  # This will create the following keys + attributes:
+  # - secret/region-one/config-demo:
+  #     secret: ...<generated basicPolicy secret>...
+  #     secretprompt: ...<as input by the user>...
+  #     secretprompt2: ...<as input by the user. If just enter is pressed it will be 'defaultvalue'>...
+  #     secretfile: ...<content of the file as input by user. If just enter is pressed the file will be /tmp/ca.crt...0
+  #     ca_crt: ...<content of /tmp/ca.crt>...
+  #     ca_crt_b64: ...<content of /tmp/ca.crt base64-encoded before uploading to vault>...
+  # - secret/snowflake.blueprints.rhecoeng.com:
+  #     secret: ...<generated basicPolicy secret>...
+  #     secretprompt: ...<as input by the user>...
+  #     secretprompt2: ...<as input by the user. If just enter is pressed it will be 'defaultvalue'>...
+  #     secretfile: ...<content of the file as input by user. If just enter is pressed the file will be /tmp/ca.crt...0
+  #     ca_crt: ...<content of /tmp/ca.crt>...
+  #     ca_crt_b64: ...<content of /tmp/ca.crt base64-encoded before uploading to vault>...
+  - name: config-demo
+    vaultMount: secret
+    vaultPrefixes:
```

Comment:
Can we do some cursory research how (for example) CyberArk structures things and potentially choose terms that are independent of the implementation?

Reply:
Docs to upload secrets were a bit hidden IMO. https://docs.conjur.org/Latest/en/Content/Developer/Conjur_API_Set_Secret.htm?TocPath=Developer%7CREST%C2%A0APIs%7C_____9 has something. I suspect that the 'path' concept of uploading a secret somewhere is similar. What I am not entirely sure, and could not figure it out from a cursory look at the docs, is, if given a path, you can have multiple attributes with secret material for each attribute.
In any case though I don't know that exposing generic terms that we have then to translate (potentially imperfectly) internally is the way to go as it is bound to be a lossy direction and a lot of code that is bound to have bugs (same with hiding a password policy behind some generic term instead of saying "this is the vaultPolicy")